core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/HeaderUnmarshaller.java

```diff
@@ -69,9 +71,17 @@ private SimpleHeaderUnmarshaller(StringToValueConverter.StringToValue<T> stringT
 
         @Override
         public T unmarshall(XmlUnmarshallerContext context, List<XmlElement> content, SdkField<T> field) {
-            return context.response().firstMatchingHeader(field.locationName())
-                          .map(s -> stringToValue.convert(s, field))
-                          .orElse(null);
+            try {
+                return context.response().firstMatchingHeader(field.locationName())
+                              .map(s -> stringToValue.convert(s, field))
+                              .orElse(null);
+            } catch (RuntimeException e) {
+                log.warn(() -> (String.format("Exception found while parsing response header {}", e.getMessage())));
```

Comment:
nit, we should probably log the entire stacktrace. 
`log.warn(() -> "Exception found while parsing response header", e));`

Reply:
hmm.. I thoght about that, but felt that this would provide the user with sufficient info. The message would be something like 'Unable to parse Expires header at index 1'. Do you think providing a trace gives more useful info ?